packages/cli/src/commands/grandamento/propose.ts

```diff
@@ -0,0 +1,63 @@
+import { StableToken } from '@celo/contractkit'
+import { flags } from '@oclif/command'
+import { BaseCommand } from '../../base'
+import { newCheckBuilder } from '../../utils/checks'
+import { displaySendTx } from '../../utils/cli'
+import { Flags } from '../../utils/command'
+import { enumEntriesDupWithLowercase } from '../../utils/helpers'
+
+const stableTokenOptions = enumEntriesDupWithLowercase(Object.entries(StableToken))
+
+export default class Propse extends BaseCommand {
+  static description = 'Proposes a Granda Mento exchange'
+
+  static flags = {
+    ...BaseCommand.flags,
+    from: Flags.address({ required: true, description: 'The address with tokens to exchange' }),
+    value: Flags.wei({
+      required: true,
+      description: 'The value of the tokens to exchange',
+    }),
+    stableToken: flags.enum({
+      required: true,
+      options: Object.keys(stableTokenOptions),
+      description: 'Name of the stable to receive or send',
+      default: 'cUSD',
+    }),
+    sellCelo: flags.boolean({
+      required: true,
+      description: 'Sell or buy CELO',
+    }),
+  }
+
+  async run() {
+    const celoToken = await this.kit.contracts.getGoldToken()
+    const grandaMento = await this.kit.contracts.getGrandaMento()
+
+    const res = this.parse(Propse)
+    const signer = res.flags.from
+    const sellAmount = res.flags.value
+    const stableToken = stableTokenOptions[res.flags.stableToken]
+    const sellCelo = res.flags.sellCelo
+
+    this.kit.defaultAccount = signer
+
+    const tokenToSell = sellCelo ? celoToken : await this.kit.contracts.getStableToken(stableToken)
+
+    await newCheckBuilder(this, signer).hasEnoughErc20(signer, sellAmount, celoToken.address)
+
+    await displaySendTx(
+      'increaseAllowance',
+      tokenToSell.increaseAllowance(grandaMento.address, sellAmount.toFixed())
+    )
+
+    await displaySendTx(
+      'propose',
+      await grandaMento.createExchangeProposal(
+        this.kit.celoTokens.getContract(stableToken),
+        sellAmount,
+        sellCelo
+      )
+    )
```

Comment:
Because there's some info that can only really be learned after proposing (like the id, buyAmount, effective price, etc) that the proposer will need to be aware of and share with the community, wdyt of printing the exchange proposal info here?

Reply:
Do you know how I actually get the proposal ID in this function?